lib/model/frames.js

```diff
@@ -97,11 +101,12 @@ const { Form } = require('./frames/form');
 Form.Attachment = class extends Frame.define(
   table('form_attachments'),
   'formId',                             'formDefId',
-  'blobId',                             'name',         readable,
-  'type',       readable,               'updatedAt',    readable
+  'blobId',                             'datasetId',
+  'name',         readable,             'type',       readable,
+  'updatedAt',    readable
 ) {
   forApi() {
-    const data = { name: this.name, type: this.type, exists: (this.blobId != null) };
+    const data = { name: this.name, type: this.type, blobExists: this.blobId != null, datasetExists: this.datasetId != null };
```

Comment:
I can see the appeal of the symmetry of `blobExists` and `dataExists`. But if we remove `exists`, that would be a breaking change to the API. What do you think about keeping `exists`, but adding a boolean `dataset` property? I'm thinking of something like:

```js
const data = { name: this.name, type: this.type, exists: this.blobId != null || this.datasetId != null, dataset: this.datasetId != null };
```

Alternatively (and this would be an additional property), but we could also consider adding `blobExists` and `datasetExists` while keeping `exists`. If there's a way to avoid removing this property, my instinct is that that'd be preferable.

Reply:
I think we should address this soon because of its impact on frontend and possibly enketo!